common/app/targeting/CampaignAgent.scala

```diff
@@ -0,0 +1,27 @@
+package targeting
+
+import common._
+import com.gu.targeting.client.CampaignCache
+import conf.Configuration
+import scala.util.control.NonFatal
+import scala.concurrent.Future
+
+object CampaignAgent extends Logging with ExecutionContexts {
+  private val agent = AkkaAgent[CampaignCache](CampaignCache(List()))
+
+  def refresh(): Future[Unit] = {
+    Configuration.targeting.campaignsUrl.map(url => {
+      CampaignCache.fetch(url).map(agent.send)
+    }).getOrElse(Future.failed(throw new BadConfigurationException("Campaigns URL not configured")))
```

Comment:
send happens to return unit, but it's actually starting an asynchronous task, so although the types happen to line up it's still effectively hiding an orphaned Future from you.

So at present the scheduler will consider the job finished even if the send is still running.  And any error from the send (if that's possible) will go somewhere undefined.

All this is unlikely to break something in the successful case, but when I was investigating perf issues last year, this kind of lost Futures problem caused me a lot of hassle.

So I'd recommend using alter instead of send, and sequencing the future that's been returned so that the scheduler knows when they're both finished successfully and can log properly.